network/common/src/main/java/org/apache/spark/network/protocol/AbstractResponseMessage.java

```diff
@@ -17,23 +17,15 @@
 
 package org.apache.spark.network.protocol;
 
-import com.google.common.base.Objects;
-import io.netty.buffer.ByteBuf;
-
 import org.apache.spark.network.buffer.ManagedBuffer;
-import org.apache.spark.network.buffer.NettyManagedBuffer;
 
 /**
- * Abstract class for response messages that contain a large data portion kept in a separate
- * buffer. These messages are treated especially by MessageEncoder.
+ * Abstract class for response messages.
  */
-public abstract class ResponseWithBody implements ResponseMessage {
-  public final ManagedBuffer body;
-  public final boolean isBodyInFrame;
+public abstract class AbstractResponseMessage extends AbstractMessage implements ResponseMessage {
```

Comment:
That would require dummy implementations in `ChunkFetchFailure`, `RpcFailure` and `StreamFailure`...

I actually don't see a lot of value in having separate interfaces for `ResponseMessage` and `RequestMessage`; this is not Scala, so you can't have a `sealed trait` and have the compiler help you when you miss something in a match. So you could just have `Message` with no need for any of the other interfaces / abstract classes. But I didn't want to do that cleanup in this change.